bitnami/argo-cd/templates/dex/metrics-svc.yaml

```diff
@@ -0,0 +1,44 @@
+{{- if and .Values.dex.enabled .Values.dex.metrics.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "argocd.dex" . }}-metrics
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: dex
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if or .Values.commonAnnotations .Values.dex.metrics.service.annotations }}
+  annotations:
+  {{- if .Values.commonAnnotations }}
+  {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+  {{- if .Values.dex.metrics.service.annotations }}
+  {{- include "common.tplvalues.render" ( dict "value" .Values.dex.metrics.service.annotations "context" $ ) | nindent 4 }}
+  {{- end }}
+  {{- end }}
+spec:
+  type: {{ .Values.dex.metrics.service.type }}
+  {{- if (or (eq .Values.dex.metrics.service.type "LoadBalancer") (eq .Values.dex.metrics.service.type "NodePort")) }}
+  externalTrafficPolicy: {{ .Values.dex.metrics.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  {{ if eq .Values.dex.metrics.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.dex.metrics.service.loadBalancerSourceRanges }}
+  {{ end }}
+  {{- if (and (eq .Values.dex.metrics.service.type "LoadBalancer") (not (empty .Values.dex.metrics.service.loadBalancerIP))) }}
+  loadBalancerIP: {{ .Values.dex.metrics.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+    - name: metrics
```

Comment:
Use the `http-` prefix to have compatibility with istio

https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/